icicle/include/icicle/hash/merkle_tree.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <memory>
+#include "icicle/common.h" // For limb and stream types
+#include "icicle/backend/hash/merkle_tree_backend.h"
+#include "icicle/hash/merkle_tree_config.h"
+
+namespace icicle {
+
+  class MerkleTree
+  {
+  public:
+    /**
+     * @brief Constructor for the MerkleTree class.
+     * @param backend Shared pointer to the backend for Merkle tree operations.
+     */
+    MerkleTree(std::shared_ptr<MerkleTreeBackend> backend) : m_backend(std::move(backend)) {}
+
+    /**
+     * @brief Build the Merkle tree from the leaves.
+     * @param leaves Pointer to the leaves of the tree.
+     * @param config Configuration struct for the merkle tree.
+     * @param secondary_leaves Pointer to the secondary_leaves in case the tree receives inputs from a secondary stream.
+     * @return Error code of type eIcicleError.
+     */
+    eIcicleError build(const limb_t* leaves, const MerkleTreeConfig& config, const limb_t* secondary_leaves = nullptr)
+    {
+      return m_backend->build(leaves, config, secondary_leaves);
+    }
+
+    /**
+     * @brief Get the root of the Merkle tree.
+     * @param root Output parameter for the root of the tree.
+     * @return Error code of type eIcicleError.
+     */
+    eIcicleError get_root(const limb_t*& root) const { return m_backend->get_root(root); }
+
+    /**
+     * @brief Get the Merkle path from a specified element index.
+     * @param leaves Pointer to the leaves of the tree.
+     * @param element_idx Index of the element for which the path is to be retrieved.
+     * @param path Output parameter for the Merkle path.
+     * @param config Configuration struct for the Merkle tree.
+     * @return Error code of type eIcicleError.
+     */
+    eIcicleError
+    get_path(const limb_t* leaves, uint64_t element_idx, limb_t* path, const MerkleTreeConfig& config) const
```

Comment:
please add input parameter 
const bool is_pruned

Reply:
I will refactor this path thing as we discussed. Also I think that the verification should be a method of the path object and not the tree since typically the verifier will call it and he doesn't have the tree but only the path and root.
Also I want to go over plonky3 and see their API to make sure this is what users need.